force-app/main/adapter/in/sobjects/contact/ContactAdapter.cls

```diff
@@ -681,13 +681,7 @@ public inherited sharing class ContactAdapter extends fflib_SObjects2 {
             // Check if the contact has Address Override enabled
             if (contact.Is_Address_Override__c == true) {
                 // Create a new address record specific to this contact
-                Address__c newAddress = new Address__c();
-                newAddress.Household_Account__c = contact.AccountId; // Link to household/account
-                newAddress.MailingStreet__c = contact.MailingStreet;
-                newAddress.MailingCity__c = contact.MailingCity;
-                newAddress.MailingState__c = contact.MailingState;
-                newAddress.MailingPostalCode__c = contact.MailingPostalCode;
-                newAddress.MailingCountry__c = contact.MailingCountry;
+                Address__c newAddress = new NPSP_Address(contact).getRecord();
```

Comment:
@lparrott could you explain a little how this is preserving the country/state picklist codes? Is it because `NPSP_Address` uses that `copyFromSObject` which checks for state/country enabled?

Reply:
That's exactly why!

Reply:
Looks like it ends up in this method, which does respect State and Country Pick Lists: https://github.com/SalesforceFoundation/NPSP/blob/main/force-app/main/default/classes/NPSP_Address.cls#L334